spring-core/src/main/java/org/springframework/util/StringUtils.java

```diff
@@ -810,8 +811,9 @@ public static TimeZone parseTimeZoneString(String timeZoneString) {
 		}
 		List<String> result = new ArrayList<>();
 		result.addAll(Arrays.asList(array1));
+		Set<String> stringSet = new HashSet<>(result);
```

Comment:
but your way is not equal to original, just becouse result can contains same items if they was present in a first array.